nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/model/impl/ArrayProperty.java

```diff
@@ -114,21 +111,16 @@ public Serializable normalize(Object value) throws PropertyConversionException {
     }
 
     protected Serializable convert(Collection<?> value) throws PropertyConversionException {
-        try {
-            Type fieldType = getType().getFieldType();
-            Collection<Object> col = new ArrayList<>(value.size());
-            for (Object v : value) {
-                if (v == null) {
-                    col.add(null);
-                } else {
-                    col.add(fieldType.convert(v));
-                }
+        Property typedProperty = getRoot().createProperty(null, getType().getField(), NONE);
+        Collection<Object> col = new ArrayList<>(value.size());
+        for (Object v : value) {
+            if (v == null) {
+                col.add(null);
```

Comment:
We shouldn't keep `null` in arrays as it's not handled very well (and not desirable) at low level. But this is for another ticket.

Reply:
Created NXP-23986 for null handling